src/clj/web/lobby.clj

```diff
@@ -523,13 +523,21 @@
         (->> (assoc lobbies gameid)))
     lobbies))
 
-(defn swap-text [players]
-  (let [[player1 player2] (mapv swap-side players)
-        player1-username (-> player1 :user :username)
-        player2-username (-> player2 :user :username)]
-    (str player1-username " has swapped sides. "
-         player1-username " is now " (:side player1) ". "
-         player2-username " is now " (:side player2) ".")))
+(defn- player-swap-text [players]
+  (apply
+   str
+   (map #(str (-> % :user :username) " is now " (-> % :side) ". ")
+        players)))
+
+(defn swap-text [players side]
+  (let [[{player-1-side :side
+          {player-1-username :username} :user}] players]
+    (str
+     player-1-username " has swapped sides. "
+     (cond
+       (= side "Any Side") "Waiting for oppenent."
+       (= side player-1-side) (player-swap-text players)
+       :else (player-swap-text (mapv swap-side players))))))
```

Comment:
I think this is overly complicated for what it's trying to do. I'm not entirely sure why it's written this way, so my apologies for not understanding. When would the code reach the `:else` block? No wait, the odd one is when the first player has chosen to swap to their own/same side? I would expect that we don't print anything from such a move.

Outside of that, I think that I would prefer this to go back to a variation on how it was originally written. Maybe something like this? If you want the destructuring, you can use it. But I don't think we need another layer of function calls.

```clojure
(defn swap-text [players current-side]
  (let [[player1 player2] (mapv swap-side players)
        player1-username (-> player1 :user :username)
        player2-username (-> player2 :user :username)]
    (str player1-username " has swapped sides. "
         (if (= side "Any Side")
           "Waiting for opponent."
           (str player1-username " is now " (:side player1) ". "))
         (when player2
           (str player2-username " is now " (:side player2) ".")))))
```

Reply:
The reason is that this is a very procedural function: take the two players, print their usernames with opposite side strings. The indirection of the `cond` and `player-swap-text` makes it much harder to see that simple string generation.

Reply:
So, this is a bit niche but the problem is mostly when there's only one person in the lobby. When there's two people in the lobby there isn't a "Any Side", since we just keep swapping back and forth.
However, if there's only one person, here's the current behaviour:

### Swap sides -> Any Side

- `locks has swapped sides. locks is now Corp. is now .` 

### Swap sides -> Corp

- `locks has swapped sides. locks is now Corp. is now .`
- Swap sides -> Corp
- `locks has swapped sides. locks is now Runner. is now .
- Swap sides -> Any Side
- `locks has swapped sides. locks is now Runner. is now .

### Swap sides -> Runner

- `locks has swapped sides. locks is now Runner. is now .`
- Swap sides -> Runner
- `locks has swapped sides. locks is now Corp. is now .`
- Swap sides -> Any Side
- `locks has swapped sides. locks is now Corp. is now .`

Reply:
Right, so my suggestion is to rely on `when` to return `nil` (which `str` converts to an empty string) if player 2 doesn't exist, and to either print "Waiting on opponent" if player 1 is "Any Side" or player 1's new side (which is set by calling `(mapv swap-side players)` already). There shouldn't be any other logic here, right?

Reply:
So if you pick the side that was already picked, we return nothing?

Reply:
Should we even show the side you currently are? Or just show the ones you're not? In what circumstances is it helpful to be "Corp" and then click "Corp" and have it say "You've swapped sides and are now Corp."?

Reply:
I definitely don't think it's useful! :P What do you think of disabling the button, so that the listing is always consistent?

Reply:
Yeah, if that's possible, let's do that. Then it simplifies our logic so we only have to deal with actually moving from Corp->Runner or Runner->Corp.